osquery/tables/system/windows/hvci_status.cpp

```diff
@@ -0,0 +1,87 @@
+#include <osquery/sql.h>
+#include <osquery/system.h>
+#include <osquery/tables.h>
+
+#include "osquery/core/conversions.h"
+#include "osquery/core/windows/wmi.h"
+
+namespace osquery {
+namespace tables {
+
+QueryData genHVCIStatus(QueryContext& context) {
+  Row r;
+  QueryData results;
+
+  const WmiRequest wmiSystemReq("SELECT * FROM Win32_DeviceGuard",
+                                (BSTR)L"ROOT\\MICROSOFT\\WINDOWS\\DEVICEGUARD");
+  const std::vector<WmiResultItem>& wmiResults = wmiSystemReq.results();
+  if (wmiResults.empty()) {
+    LOG(WARNING) << "Error retreiving information from WMI.";
+    return results;
+  }
+  for (const auto& data : wmiResults) {
+    long vbsmethod;
+    long codepolicystatusmethod;
+    long umcipolicymethod;
+
+    data.GetString("Version", r["version"]);
+    data.GetString("InstanceIdentifier", r["instance_identifier"]);
+    data.GetLong("VirtualizationBasedSecurityStatus", vbsmethod);
+    data.GetLong("CodeIntegrityPolicyEnforcementStatus",
+                 codepolicystatusmethod);
+    data.GetLong("UsermodeCodeIntegrityPolicyEnforcementStatus",
+                 umcipolicymethod);
+
+    std::string vbsmethod_str;
+    std::map<long, std::string> vbsmethods;
+
+    std::string codepolicystatusmethod_str;
+    std::map<long, std::string> codepolicystatusmethods;
+
+    std::string umcipolicymethod_str;
+    std::map<long, std::string> umcipolicymethods;
+
+    vbsmethods[0] = "VBS_NOT_ENABLED";
+    vbsmethods[1] = "VBS_ENABLED_AND_NOT_RUNNING";
+    vbsmethods[2] = "VBS_ENABLED_AND_RUNNING";
+
+    codepolicystatusmethods[0] = "OFF";
+    codepolicystatusmethods[1] = "AUDIT_MODE";
+    codepolicystatusmethods[2] = "ENFORCED_MODE";
+
+    umcipolicymethods[0] = "OFF";
+    umcipolicymethods[1] = "AUDIT_MODE";
+    umcipolicymethods[2] = "ENFORCED_MODE";
+
+    if (vbsmethods.find(vbsmethod) != vbsmethods.end()) {
```

Comment:
Let's save this value so that you don't grab it twice. In fact you could just make this a ternary:
```
auto vbs_meth = vbsmethods.find(vbsmethod);
r["vbs_status"] = vbs_meth != vbs_methods.end() ? vbs_meth->second : "UNKNOWN";
```
Here and throughout.

Reply:
Fixed